web/pgadmin/browser/server_groups/servers/databases/publications/__init__.py

```diff
@@ -710,38 +868,97 @@ def get_tables(self, gid, sid, did):
                               server_type=self.manager.server_type
                               )
         status, rset = self.conn.execute_2darray(sql)
+        if not status:
+            return internal_server_error(errormsg=res)
+
+        if self.manager.version < 150000:
+            for row in rset['rows']:
+                res.append(
+                    {
+                        'label': row['table'],
+                        'value': row['table']
+                    }
+                )
+        elif self.manager.version >= 150000:
+            for row in rset['rows']:
+                res.append(
+                    {
+                        'label': row['table'],
+                        'value': row['table'],
+                        'tid': row['tid']
+                    }
+                )
+        return make_json_response(
+            data=res,
+            status=200
+        )
+
+    @check_precondition
+    def get_all_columns(self, gid, sid, did):
+        """
+        This function returns the columns list.
+
+        Args:
+            gid: Server Group ID
+            sid: Server ID
+            did: Database ID
+            tid: Table ID
+        """
+        data = request.args
+
+        res = []
+
+        sql = render_template("/".join([self.template_path,
+                                        'get_all_columns.sql']),
+                              tid=data['tid'], conn=self.conn)
+        status, rset = self.conn.execute_2darray(sql)
         if not status:
             return internal_server_error(errormsg=res)
         for row in rset['rows']:
             res.append(
                 {
-                    'label': row['table'],
-                    'value': row['table']
+                    'label': row['column'],
+                    'value': row['column'],
                 }
             )
         return make_json_response(
             data=res,
             status=200
         )
 
-    def _get_old_table_data(self, pname, res):
+    def _get_old_table_data(self, pname, res, exempt_schema_list=[]):
         """
         This function return table details before update
         :param pname:
         :param res:
         :return:old_data
         """
-
-        table_sql = render_template(
-            "/".join([self.template_path, self._GET_TABLE_FOR_PUBLICATION]),
-            pname=pname
-        )
+        if self.manager.version < 150000:
+            table_sql = render_template(
+                "/".join([self.template_path,
+                          self._GET_TABLE_FOR_PUBLICATION]),
+                pname=pname
```

Comment:
Template data can be created based on conditions and then passed as `**data` to avoid code repeat.